azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1459 @@
+
+		result := frontdoor.LoadBalancingSettingsModel{
+			ID:   id,
+			Name: utils.String(name),
+			LoadBalancingSettingsProperties: &frontdoor.LoadBalancingSettingsProperties{
+				SampleSize:                    utils.Int32(sampleSize),
+				SuccessfulSamplesRequired:     utils.Int32(successfulSamplesRequired),
+				AdditionalLatencyMilliseconds: utils.Int32(additionalLatencyMilliseconds),
+			},
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorRoutingRule(input []interface{}, subscriptionId string, resourceGroup, frontDoorName string) *[]frontdoor.RoutingRule {
+	if len(input) == 0 {
+		return nil
+	}
+
+	output := make([]frontdoor.RoutingRule, 0)
+
+	for _, rr := range input {
+		routingRule := rr.(map[string]interface{})
+
+		id := routingRule["id"].(string)
+		frontendEndpoints := routingRule["frontend_endpoints"].([]interface{})
+		acceptedProtocols := routingRule["accepted_protocols"].([]interface{})
+		ptm := routingRule["patterns_to_match"].([]interface{})
+		enabled := routingRule["enabled"].(bool)
+		name := routingRule["name"].(string)
+
+		patternsToMatch := make([]string, 0)
+
+		for _, p := range ptm {
+			patternsToMatch = append(patternsToMatch, p.(string))
+		}
+
+		var routingConfiguration frontdoor.BasicRouteConfiguration
+
+		if rc := routingRule["redirect_configuration"].([]interface{}); len(rc) != 0 {
+			routingConfiguration = expandArmFrontDoorRedirectConfiguration(rc)
+		} else if fc := routingRule["forwarding_configuration"].([]interface{}); len(fc) != 0 {
+			routingConfiguration = expandArmFrontDoorForwardingConfiguration(fc, subscriptionId, resourceGroup, frontDoorName)
+		}
+
+		currentRoutingRule := frontdoor.RoutingRule{
+			ID:   utils.String(id),
+			Name: utils.String(name),
+			RoutingRuleProperties: &frontdoor.RoutingRuleProperties{
+				//ResourceState:
+				FrontendEndpoints:  expandArmFrontDoorFrontEndEndpoints(frontendEndpoints, subscriptionId, resourceGroup, frontDoorName),
+				AcceptedProtocols:  expandArmFrontDoorAcceptedProtocols(acceptedProtocols),
+				PatternsToMatch:    &patternsToMatch,
+				EnabledState:       frontdoor.RoutingRuleEnabledState(expandArmFrontDoorEnabledState(enabled)),
+				RouteConfiguration: routingConfiguration,
+			},
+		}
+
+		output = append(output, currentRoutingRule)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorAcceptedProtocols(input []interface{}) *[]frontdoor.Protocol {
+	if len(input) == 0 {
+		return &[]frontdoor.Protocol{}
+	}
+
+	output := make([]frontdoor.Protocol, 0)
+
+	for _, ap := range input {
+		result := frontdoor.HTTPS
+
+		if ap.(string) == string(frontdoor.HTTP) {
+			result = frontdoor.HTTP
+		}
+
+		output = append(output, result)
+	}
+
+	return &output
+}
+
+func expandArmFrontDoorSubResource(subscriptionId string, resourceGroup string, frontDoorName string, resourceType string, resourceName string) *frontdoor.SubResource {
+	result := frontdoor.SubResource{
```

Comment:
Instead of having this function could we just assign `frontdoor.SubResource` inline so its clear what is going on?

Reply:
Sure, fixed.